File: frontend/src/app/cart/domain/order.ts
export class Order {
  totalPrice: number;
  itemIds: Array<number>;
  email: string;


  constructor(totalPrice: number, itemIds: Array<number>, email: string) {
    this.totalPrice = totalPrice;
    this.itemIds = itemIds;
    this.email = email;
  }
}

File: backend/src/main/java/com/example/backend/order/domain/Order.java
package com.example.backend.order.domain;

import java.util.Collections;
import java.util.List;

public class Order {

    private List<Long> itemIds;
    private String email;
    private Double totalPrice;

    public Order(List<Long> itemIds, String email, Double totalPrice)
    {
        this.itemIds = itemIds == null ? Collections.emptyList() : itemIds;
        this.email = email;
        this.totalPrice = totalPrice;
    }

    public List<Long> getItemIds()
    {
        return itemIds;
    }

    public String getEmail()
    {
        return email;
    }

    public Double getTotalPrice()
    {
        return totalPrice;
    }

    @Override
    public String toString()
    {
        return "Order{" +
            "itemIds=" + itemIds +
            ", email='" + email + '\'' +
            ", totalPrice=" + totalPrice +
            '}';
    }
}
